### ISSUE TITLE
Extension authoring - Live updates without saving

### ISSUE THREAD

**Issue by XVincentX** [Microsoft/vscode#24307]
Hello guys,
I am working on my extension and I am struggling with the API as I have no idea what's the best way to implement my use case - perhaps you could give me some help?

Basically I am trying to offer almost the same experience of having a Markdown document with preview opened on the side: as you type into the new document, you get the preview updated.

Thing is that actually this feature you're offering it's working only if the file is saved on the disk. I would like to offer the same experience for untitled files. What would be the best way to accomplish this?

The way I was trying to do this is to install a global handler on the client side to all document changes event and then try to verify if my preview is somehow opened:

```javascript
  workspace.onDidChangeTextDocument((changeEvent) => {
    if ((changeEvent.document.languageId === 'apiblueprint' || changeEvent.document.languageId === 'swagger') &&
      workspace.textDocuments.some((doc) => doc.isUntitled && doc.fileName.includes('parseResult.json'))
    ) {
      
    }
  });

```
Basically checking that:
1. The changed document opened has my language ID
2. There's a preview opened (using the untitled filename I am creating)

but to be honest it does not seem to be that clever to me.

My extension is a server/client thing, so I can work on both sides to make this happen.

**Comment by jrieken** [comment]
> I am working on my extension and I am struggling with the API as I have no idea what's the best way to implement my use case - perhaps you could give me some help?

Take a look at our content provider and the `onDidChange` event it exposes: https://github.com/Microsoft/vscode/blob/master/extensions/markdown/src/previewContentProvider.ts#L38

> Basically I am trying to offer almost the same experience of having a Markdown document with preview opened on the side: as you type into the new document, you get the preview update

I don't think that's correct but @mjbvz will know
